pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,47 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/sirupsen/logrus"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&RootCA{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	ca := &RootCA{}
+	dependencies.Get(ca)
+
+	cfg := &CertCfg{
+		Subject:      pkix.Name{CommonName: "ironic-virtual-media", Organization: []string{"OpenShift Bootstrap"}},
```

Comment:
In CBO this is set to the Provisioning VIP instead of "ironic-virtual-media", and with no Organization.
The Provisioning VIP is also present in the `IPAddresses` field.
If there is no Provisioning VIP set (i.e. there is no provisioning network), `"localhost"` is used (this seems like a mistake, but whatevs).

The externalURL at least is pointing to the API VIP (which will at least be _a_ valid IP address of the bootstrap VM during bootstrapping, although you'd think this would require an address on the provisioning network if it is not disabled?). So maybe the API VIP is the IP we need to use in this case.

Reply:
In the end, it all does not matter because we'll need to disable the host validation.. but yeah, strictly speaking you're right: it should be either the provisioning IP or the API VIP.

Localhost in CBO is a historical artefact from the times when we could not predict the IP without a provisioning network.

Reply:
Note for Mahnoor: we'd need to depend on the InstallConfig asset and get the IPs from it.

Reply:
Done

Reply:
> In CBO this is set to the Provisioning VIP instead of "ironic-virtual-media", and with no Organization.

Follow-up: seems like the validation here is much more picky. We had to bring Organization back. Fortunately, neither CommonName nor Organization matter here at all.